### ISSUE TITLE
RFE: html result as single file

### ISSUE THREAD

**Issue by jscotka** [avocado-framework/avocado#2004]
Comprimation of files causes that gz files are not loaded, so that CSS and JS is not loaded,
see taskotron: 
https://taskotron.fedoraproject.org/artifacts/all/709a28a4-3a19-11e7-b258-5254008e42f6/task_output/avocado-result/html/results.html
Discussed with @ldoktor personally.

**Comment by ldoktor** [comment]
Let me rephrase it a bit. Basically some results dbs (Jenkins/taskotron) are storing the artifacts compressed and always uncompress the single file they are asked for, making the html results hideous. Instead of a normal file with includes we should generate single multi-part html file and I think it's a good idea.

**Comment by clebergnu** [comment]
If it's at all possible, then yes, +1 here.

**Comment by ldoktor** [comment]
It is possible:
[results.zip](https://github.com/avocado-framework/avocado/files/1077800/results.zip)

Do you see this as a new feature, or can it be included in this release? (PS: This one was created manually, but it is really simple to put it directly in html plugin)

**Comment by lmr** [comment]
Curiously, I can see the HTML file generated just fine here on Chrome 59.0.3071.86 (granted, right now I'm on the macOS laptop).

**Comment by lmr** [comment]
Having a single file HTML was considered initially, but it seemed excessively ugly having a single file with all assets. Also I'm not sure all assets can be inlined in a way that further uglifies the HTML file.

However I can see the appeal of a single file, so by all means, go ahead.

**Comment by lmr** [comment]
![screenshot 2017-06-15 16 21 42](https://user-images.githubusercontent.com/296807/27185811-bd0515f2-51e6-11e7-9eb3-aceecbaa891e.png)

**Comment by lmr** [comment]
@ldoktor I just saw your inlined file, it works fine. It's not pretty on the inside, but it does deliver!

**Comment by ldoktor** [comment]
Well the cons is it looks ugly, but you have `results.{json,tap}` to get vim-readable results. Also you are free to ignore the <header> section and the rest is intact.

As for the changes, basically I replaced the `<link href="css/bootstrap.min.css" rel="stylesheet">` with `<style>...</style>`, the `<script src="js/jquery.min.js"></script>` with `<script>...</script>` inside the `<header>` section and embedded the 3 pictures we have in `css` files using `base64`. All of those features are actually simpler forms than external links, so it should render well everywhere.

**Comment by lmr** [comment]
Yep, I just thought strange that my browser can display the URL that @jscotka reported perfectly, no additional steps required, so I'm taking with some other browsers or browser versions that does not happen. I think it's going to be fine to inline everything.

**Comment by ldoktor** [comment]
Oh, I thought you are talking about mine version. Well you are right, they must have changed configuration as most files are compressed, but when you look at https://taskotron.fedoraproject.org/artifacts/all/709a28a4-3a19-11e7-b258-5254008e42f6/task_output/avocado-result/html/ it's not compressing `.js`, nor `.css` files.

**Comment by ldoktor** [comment]
https://github.com/avocado-framework/avocado/pull/2072

**Comment by clebergnu** [comment]
I see all files under https://taskotron.fedoraproject.org/artifacts/all/709a28a4-3a19-11e7-b258-5254008e42f6/task_output/avocado-result/html/ ending in `.gz`.

Still, that should not be an issue, as their webservers will serve uncompressed versions of these files transparently.  Just try out the following links:
 * https://taskotron.fedoraproject.org/artifacts/all/709a28a4-3a19-11e7-b258-5254008e42f6/task_output/avocado-result/html/results.html
 * https://taskotron.fedoraproject.org/artifacts/all/709a28a4-3a19-11e7-b258-5254008e42f6/task_output/avocado-result/html/css/bootstrap.min.css

**Comment by ldoktor** [comment]
Well I said this on IRC, so let's mention it here as well. This is only true for properly configured `taskotron`. When this issue was raised, this was not enabled/supported and it is still not supported on Jenkins servers and who knows where else, therefor I'd support the single-page results... Anyway PR is pending, so let's move the discussion there...

**Comment by ldoktor** [comment]
https://github.com/avocado-framework/avocado/pull/2072 was merged, results are now available in a single html file.
